app/scripts/modules/core/src/pipeline/config/stages/bakeManifest/BakeManifestStageForm.tsx

```diff
@@ -22,6 +24,16 @@ export function BakeManifestStageForm({
 }: IBakeManifestStageFormProps & IFormikStageConfigInjectedProps) {
   const stage = formik.values;
 
+  React.useEffect(() => {
+    if (!validateProducedArtifacts(stage.expectedArtifacts)) {
+      const defaultProducedArtifact = ExpectedArtifactService.createEmptyArtifact();
+      defaultProducedArtifact.matchArtifact.type = BASE_64_ARTIFACT_TYPE;
+      defaultProducedArtifact.matchArtifact.artifactAccount = BASE_64_ARTIFACT_ACCOUNT;
+      defaultProducedArtifact.matchArtifact.customKind = false;
+      formik.setFieldValue('expectedArtifacts', [defaultProducedArtifact]);
+    }
+  }, []);
```

Comment:
Am I understanding correctly that this empty array makes it run only on the initial render, but not on re-renders?

Reply:
It'll run only once, but _after_ the initial render (can roughly map to `componentDidMount` for class components).